src/test/java/bsh/InterpreterTest.java

```diff
@@ -52,4 +55,61 @@ public void check_system_object() throws Exception {
         assertEquals(Boolean.FALSE, TestUtil.eval("return bsh.system.shutdownOnExit;"));
     }
 
+    @Test
+    public void interrupt_on_tokenizing() throws Exception {
+        final StringReader in = new StringReader("print(\"\n");
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        Interpreter bsh = new Interpreter(in, new PrintStream(baos), System.err, false);
+
+        Thread t = new Thread(bsh);
+        t.start();
+
+        Thread.sleep(250); // give Interpreter eom time to set itself up
+
+        //
+        // now the interpreter should be on a blocking read; we interrupt it,
+        // give it some time to make the magic and make sure it tells us about it
+        //
+        bsh.interrupt(); Thread.sleep(250);
+
+        assertTrue(baos.toString().contains("... interrupted ..."));
+        assertTrue(!baos.toString().contains("Parser Error:"));
+
```

Comment:
So basically interrupted is used to avoid parser errors? The new implementation set EOF on close and allows readLine (previously isEOF) to wait for closing thread. This still allows errors to be reported if the interruption was not successful instead of just ignoring it to a flag.

Reply:
interrupted will be to tell interpreter at any level that the current input or processing should be interrupted. see my first comment on why we can not use eof (unless now Line() has a different behavior). Let me know if not clear.

Reply:
Yes I understand why you are using `boolean interrupted` but all it is is a flag, it does not do anything nor does it guarantee anything. `Thread.interrupted` guarantees that the Thread will end soon. `EOF` means we have reached EOF and can abort the repl, they are both truthy and falsey but they already have tied to functionality. So why use something else which yet has no functionality other than true or false?